test/dandelion_voting_disputable.js

```diff
@@ -0,0 +1,373 @@
+        const beforeVoteEndBlock = voteDurationBlocks.sub(bn(4)) // 4 blocks since vote created
+        await voting.mockAdvanceBlocks(beforeVoteEndBlock)
+
+        const { isOpen: isOpenBeforeEndDate } = await getVoteState(voting, voteId)
+        assert.isTrue(isOpenBeforeEndDate, 'vote is not open before end date')
+
+        await voting.mockAdvanceBlocks(1)
+
+        const { isOpen: isOpenAtVoteEndDate } = await getVoteState(voting, voteId)
+        assert.isFalse(isOpenAtVoteEndDate, 'vote is open at end date')
+
+        await voting.mockAdvanceBlocks(1)
+
+        const { isOpen: isOpenAtAfterEndDate } = await getVoteState(voting, voteId)
+        assert.isFalse(isOpenAtAfterEndDate, 'vote is open after end date')
+      })
+    }
+
+    context('when allowed', () => {
+      beforeEach('dispute and allow vote', async () => {
+        await agreement.dispute({ actionId })
+        await agreement.executeRuling({ actionId, ruling: RULINGS.IN_FAVOR_OF_SUBMITTER })
+      })
+
+      it('canChallenge returns true', async () => {
+        assert.isTrue(await voting.canChallenge(voteId))
+      })
+
+      it('canClose returns false', async () => {
+        assert.isFalse(await voting.canClose(voteId))
+      })
+
+      itResumesTheVote()
+    })
+
+    context('when voided', () => {
+      beforeEach('dispute and void vote', async () => {
+        await agreement.dispute({ actionId })
+        await agreement.executeRuling({ actionId, ruling: RULINGS.REFUSED })
+      })
+
+      itResumesTheVote()
+    })
+  })
+
+  describe('cancelled', () => {
+    let pauseBlock, currentBlock
+
+    beforeEach('create vote and challenge', async () => {
+      await createVote({ voter: voter51, cast: false })
+      await agreement.challenge({ actionId })
+      pauseBlock = await voting.getBlockNumberPublic()
+
+      await voting.mockAdvanceBlocks(bn(ONE_DAY_BLOCKS))
+      currentBlock = await voting.getBlockNumberPublic()
+    })
+
+    const itCancelsTheVote = () => {
+      it('cancels the vote', async () => {
+        const expectedPauseDuration = currentBlock.sub(pauseBlock)
+        const { actionId: voteActionId, pausedAtBlock, pauseDurationBlocks, status } = await voting.getDisputableInfo(voteId)
+        assertBn(status, VOTE_STATUS.CANCELLED, 'vote status does not match')
+
+        assertBn(voteActionId, actionId, 'action ID does not match')
+        assertBn(pausedAtBlock, 0, 'paused at does not match')
+        assertBn(pauseDurationBlocks, expectedPauseDuration, 'pause duration does not match')
+      })
+
+      it('does not allow a voter to vote', async () => {
+        assert.isFalse(await voting.canVote(voteId, voter49), 'voter can vote')
+
+        await assertRevert(voting.vote(voteId, false, { from: voter49 }), 'DANDELION_VOTING_CANNOT_VOTE')
+      })
+
+      it('does not allow to execute the vote', async () => {
+        assert.isFalse(await voting.canExecute(voteId), 'voting can be executed')
```

Comment:
Should we also advance blocks here? 

```suggestion
        await voting.mockAdvanceBlocks(VOTING_DURATION_BLOCKS + EXECUTION_DELAY_BLOCKS)
        assert.isFalse(await voting.canExecute(voteId), 'voting can be executed')
```

Reply:
Good spot!